workers/task-driver/src/driver.rs

```diff
@@ -201,49 +194,89 @@ impl TaskExecutor {
         Ok(())
     }
 
+    /// Handle a request to run a task immediately
+    pub fn handle_run_immediate(
+        &self,
+        wallet_id: WalletIdentifier,
+        task_id: TaskIdentifier,
+        task: TaskDescriptor,
+    ) -> Result<(), TaskDriverError> {
+        // Check if any non-preemptable tasks conflict with this task
+        if let Some(conflicting_task) = self.state().current_committed_task(&wallet_id)? {
+            log::error!(
+                "run immediate conflicts with committed task {conflicting_task:?}, aborting..."
+            );
+            return Ok(());
```

Comment:
should we also pop the associated task from the queue, since it cannot be run?

Reply:
there is no associated task, the idea is to bypass the queue entirely